BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/MainActivity.java

```diff
@@ -122,5 +132,30 @@ public void onPointOfInterestClick(PointOfInterest pointOfInterest) {
         startActivity(intent);
     }
 
+    private void showEditDialog() {
+        android.support.v4.app.FragmentManager fm = getSupportFragmentManager();
+        CategoryDialogFragment categoryDialogFragment = new CategoryDialogFragment();
+        categoryDialogFragment.show(fm, "insert_category_name");
+    }
+
 
+    @Override
+    public void onFinishEditDialog(String categoryNameInput) {
+
+        String categoryName = categoryNameInput;
+        Boolean categoryExists = BlocSpotApplication.getSharedDataSource().categoryExists(categoryName);
+        int color = UIUtils.generateRandomColor(toolbar.getResources().getColor(android.R.color.white));
+        Category category = new Category(categoryName, color);
+        BlocSpotApplication.getSharedDataSource().insertCategoryToDatabase(category);
```

Comment:
So you insert the category here, but also a [couple of lines](https://github.com/i-zor066/BlocSpot/pull/4/files#diff-3d484af121a0be9fcfd009072e4baa65R154) below. I suspect that what you wanted to do is to insert it only if it doesn't exist yet. If that's the case than this insertion here is wrong and should be removed.